metrics/prometheus/prometheus.go

```
@@ -191,6 +191,13 @@ func metricFamily(registry Registry, name string) (mf *dto.MetricFamily, err err
 				},
 			}},
 		}, nil
+
+	case metrics.NilCounter, metrics.NilCounterFloat64, metrics.NilEWMA,
+		metrics.NilGauge, metrics.NilGaugeFloat64, metrics.NilGaugeInfo,
+		metrics.NilHealthcheck, metrics.NilHistogram, metrics.NilMeter,
+		metrics.NilResettingTimer, metrics.NilSample, metrics.NilTimer:
+		return nil, fmt.Errorf("%w: %q metric is nil", errMetricSkip, name)
+
```

Awesome 👍 For context, see [this](https://github.com/ava-labs/coreth/pull/745#discussion_r1913462161) - previously it was silently discarding them, which is quite a problem.

I'll add your case block to subnet-evm as well.

---

Case is at the wrong place, moving this one up as the first case in the switch. Gopls warnings:

```
unreachable case clause: github.com/ava-labs/libevm/metrics.Counter will always match before github.com/ava-labs/libevm/metrics.NilCounterSA4020[default](https://staticcheck.dev/docs/checks/#SA4020)
unreachable case clause: github.com/ava-labs/libevm/metrics.CounterFloat64 will always match before github.com/ava-labs/libevm/metrics.NilCounterFloat64SA4020[default](https://staticcheck.dev/docs/checks/#SA4020)
unreachable case clause: github.com/ava-labs/libevm/metrics.Gauge will always match before github.com/ava-labs/libevm/metrics.NilGaugeSA4020[default](https://staticcheck.dev/docs/checks/#SA4020)
unreachable case clause: github.com/ava-labs/libevm/metrics.GaugeFloat64 will always match before github.com/ava-labs/libevm/metrics.NilGaugeFloat64SA4020[default](https://staticcheck.dev/docs/checks/#SA4020)
unreachable case clause: github.com/ava-labs/libevm/metrics.Histogram will always match before github.com/ava-labs/libevm/metrics.NilHistogramSA4020[default](https://staticcheck.dev/docs/checks/#SA4020)
unreachable case clause: github.com/ava-labs/libevm/metrics.Meter will always match before github.com/ava-labs/libevm/metrics.NilMeterSA4020[default](https://staticcheck.dev/docs/checks/#SA4020)
unreachable case clause: github.com/ava-labs/libevm/metrics.ResettingTimer will always match before github.com/ava-labs/libevm/metrics.NilResettingTimerSA4020[default](https://staticcheck.dev/docs/checks/#SA4020)
unreachable case clause: github.com/ava-labs/libevm/metrics.Timer will always match before github.com/ava-labs/libevm/metrics.NilTimerSA4020[default](https://staticcheck.dev/docs/checks/#SA4020)
```